@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -220,12 +224,30 @@ export class TileGeometryCreator {
             }
             return filter(technique);
         };
+        this.createTextElements(tile, decodedTile, textFilter);
 
-        this.preparePois(tile, decodedTile);
+        this.createLabelRejectionElements(tile, decodedTile);
+    }
 
-        // TextElements do not get their geometry created by Tile, but are managed on a
-        // higher level.
-        this.createTextElements(tile, decodedTile, textFilter);
+    createLabelRejectionElements(tile: Tile, decodedTile: DecodedTile) {
+        if (decodedTile.pathGeometries === undefined) {
+            return;
+        }
+        const path: THREE.Vector3[] = [];
+        const point = new THREE.Vector3();
+        for (const flatpath of decodedTile.pathGeometries) {
```

Comment:
`flatpath` is a bit weird name b/c `path` is also flat. Couldn't `decodedTile.pathGeometries` be in world space already so we would save this conversion in the main thread?

Reply:
Agreed, I will rename.

Agree it makes sense to do this in the decoder. I will fix.